### PR TITLE
Add a has_exited field in validator details API

### PR DESCRIPTION
**Pull request by LefterisJP**

### CODE REVIEW

File: rotkehlchen/db/eth2.py [hunk 1]
```diff
@@ -150,6 +150,16 @@ def get_validators(self, cursor: 'DBCursor') -> list[Eth2Validator]:
         cursor.execute('SELECT * from eth2_validators;')
         return [Eth2Validator.deserialize_from_db(x) for x in cursor]
 
+    def get_exited_validator_indices(self, cursor: 'DBCursor') -> list[Eth2Validator]:
+        """Returns the indices of the tracked validators that we know have exited
```
Comment (yabirgb) [inline rotkehlchen/db/eth2.py:154]: Unless I'm still dreaming 

```suggestion
    def get_exited_validator_indices(self, cursor: 'DBCursor') -> list[int]:
        """Returns the indices of the tracked validators that we know have exited
```

Comment (LefterisJP) [reply rotkehlchen/db/eth2.py:154]: Weird that mypy did not catch this

File: rotkehlchen/db/eth2.py [hunk 2]
```diff
@@ -150,6 +150,16 @@ def get_validators(self, cursor: 'DBCursor') -> list[Eth2Validator]:
...
+
+        Does so by processing events, so will only return a list as up to date as the events we got
+        """
+        cursor.execute(
+            'SELECT validator_index FROM eth_staking_events_info WHERE is_exit_or_blocknumber=1',
+        )  # checking against literal 1 is safe since block 1 was not mined during PoS
+        return [x[0] for x in cursor]
```
[the remark this answers is not in the document]
Comment (LefterisJP) [reply rotkehlchen/db/eth2.py:161]: yes having a set is fine

**Review by yabirgb — APPROVED** [review]
Failing test is btc api. LGTM

**Pull request merged**